### DISCUSSION TITLE
How to fetch data only once on app startup?

### DISCUSSION THREAD

**Discussion by darylhjd** [vercel/next.js#54541] [Help]
### Summary

Hello. I have a use case where I want to load a user's data into the global state just once on app startup. For example, I can check whether the user is logged in or not by pinging an API, and then depending on the result of the API, I load either the logged app or the guest app.

Coming from Flutter, this was easy because Flutter's riverpod allowed us to do async functions and store state into providers before the app was actually run.

On NextJS 13, I can't figure out how to make this work. I have zustand as the global state manager, but I am not sure how to make an asynchronous call to the API server to load data into the state before the components are rendered on the client.

**Comment by Hartaithan** [comment]
@darylhjd you find solution?

**Reply by darylhjd** [reply]
I wrapped the data fetching into a provider react node and wrapped my children nodes in it in `app/layout.tsx`. Used zustand to store the fetched data.
